src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -158,6 +156,15 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
     const auto path = QString(QString::fromWCharArray(callbackInfo->VolumeDosName) + QString::fromWCharArray(callbackInfo->NormalizedPath));
     const auto requestId = QString::number(callbackInfo->TransferKey.QuadPart, 16);
 
+    if (!callbackInfo->ProcessInfo) {
+        qCCritical(lcCfApiWrapper) << "Callback parameters did not contain required process info required for the implicit hydration check, aborting" << path << requestId;
```

Comment:
for completeness self hydration should no longer happen
https://github.com/nextcloud/desktop/blob/master/src/libsync/vfs/cfapi/cfapiwrapper.cpp#L723
up for a follow up PR to remove the check ?
I still think the logs are very very valuable when one tries to understand why an hydration request occurs

Reply:
sure, will create a PR for that (though from my POV it's not a must-have for 3.17.x) :>